packages/server/events-processing-platform/ai/openai.go

```diff
@@ -0,0 +1,218 @@
+package ai
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/google/go-querystring/query"
+)
+
+const completionsUrl = "https://api.openai.com/v1/chat/completions"
+
+// Error is the error standard response from the API
+type Error struct {
+	Message string      `json:"message,omitempty"`
+	Type    string      `json:"type,omitempty"`
+	Param   interface{} `json:"param,omitempty"`
+	Code    interface{} `json:"code,omitempty"`
+}
+
+func (e *Error) Error() string {
+	return fmt.Sprintf("%s - %s", e.Code, e.Message)
+}
+
+type HttpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type OpenAiClient struct {
+	client       HttpClient
+	apiKey       string
+	organization string
+	model        string
+}
+
+// NewOpenAiClient creates a new client
+func NewOpenAiClient(apiKey, organization, model string) *OpenAiClient {
```

Comment:
yeah I thought about that when you showed it to me -- I had already written this code by then, so decided to have this be reviewed, in case it made sense to use it like this, rather then with the wrapper service.

I think this approach is more direct/simpler but again, if the wrapper service is needed i can refactor it into this (but if not, we can reduce our services by 1!)

we could also have a separate 'AI clients' module for reusability, if that is a concern

Reply:
let's build the ai module in the project and drop the other services. please move the Anthropic + OpenAi client code to this new module